content/en/docs/collector/build-connector.md

```diff
@@ -0,0 +1,567 @@
+        }, nil
+    }
+    ```
+
+    The `newConnector` function is a factory function to create an instance of a
+    connector.
+
+3.  Implement `Capabilities` method to properly implement the interface
+
+    ```go
+    // Capabilities implements the consumer interface.
+    func (c *connectorImp) Capabilities() consumer.Capabilities {
+        return consumer.Capabilities{MutatesData: false}
+    }
+    ```
+
+    Implement the `Capabilities` method to ensure your connector is of type
+    consumer. This method defines the capabilities of the component, whether the
+    component can mutate data or not. If `MutatesData` is set to true the
+    processor does modify the data or does not make a copy of data before
+    manipulation.
+
+4.  Implement `Consumer` method to consume telemetry data
+
+    ```go
+    // ConsumeTraces method is called for each instance of a trace sent to the connector
+    func (c *connectorImp) ConsumeTraces(ctx context.Context, td ptrace.Traces) error{
+    // loop through the levels of spans of the one trace consumed
+        for i := 0; i < td.ResourceSpans().Len(); i++ {
+            resourceSpan := td.ResourceSpans().At(i)
+
+            for j := 0; j < resourceSpan.ScopeSpans().Len(); j++ {
+                scopeSpan := resourceSpan.ScopeSpans().At(j)
+
+                for k := 0; k < scopeSpan.Spans().Len(); k++ {
+                    span := scopeSpan.Spans().At(k)
+                    attrs := span.Attributes()
+                    mapping := attrs.AsRaw()
+                    for key, _ := range mapping {
+                        if key == c.config.AttributeName {
+                            // create metric only if span of trace had the specific attribute
+                            metrics := pmetric.NewMetrics()
+                            return c.metricsConsumer.ConsumeMetrics(ctx, metrics)
+                        }
+                    }
+                }
+            }
+        }
+        return nil
+    }
+    ```
+
+5.  Implement `Start` and `Shutdown` methods to properly implement the interface
+
+    ```go
+    // Start implements the component.Component interface.
+    func (p *connectorImp) Start(ctx context.Context, _ component.Host) error {
+        p.logger.Info("Starting spanmetrics connector")
+        return nil
+    }
+
+    // Shutdown implements the component.Component interface.
+    func (p *connectorImp) Shutdown(context.Context) error {
+        p.logger.Info("Shutting down spanmetrics connector")
+        return nil
+    }
+    ```
+
+    Implement the `Start` and `Shutdown` methods to ensure your connector is of
+    type `component.Component`.
+
+The complete connector file:
+
+```go
+package exampleconnector
+
+import (
+    "context"
+    "fmt"
+
+    "go.opentelemetry.io/collector/component"
+    "go.opentelemetry.io/collector/consumer"
+    "go.opentelemetry.io/collector/pdata/pmetric"
+    "go.opentelemetry.io/collector/pdata/ptrace"
+    "go.uber.org/zap"
```

Comment:
nit, but our CI would fail this so why not?
```suggestion
    "go.uber.org/zap"

    "go.opentelemetry.io/collector/component"
    "go.opentelemetry.io/collector/consumer"
    "go.opentelemetry.io/collector/pdata/pmetric"
    "go.opentelemetry.io/collector/pdata/ptrace"
```